app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,51 @@
+class MarketsController < ApplicationController
+  def index
+
+  end
+
+  def new
+    @market = Market.new
+  end
+
+  def create
+    @market = Market.new(market_create_params)
+    if @market.save
+      redirect_to application_path(@market.id)
+    else
+      # @artist = Artist.select(:id, :name).order(:name)
+      # render :index
+    end
+  end
+
+  def  search
+    @markets = Market.find(params[:q])
+  end
+
+  def edit
+    @market = Market.find(params[:id])
+  end
+
+  def update
+    @market = Market.find(params[:id])
+    @market.update_attributes(market_update_params)
+    redirect_to root_path
+  end
+
+  def destroy
+    @market = Market.find(params[:id])
```

Comment:
You could pull this out this line into a `before_action` only for the controller methods that utilize it and DRY up your code a bit, if you feel it wouldn't impact clarity.